lib/ex_gram/updates/webhook.ex

```diff
@@ -68,23 +68,32 @@ defmodule ExGram.Updates.Webhook do
     config = ExGram.Config.get(:ex_gram, :webhook, []) |> Keyword.merge(opts)
     params = webhook_params(config)
 
-    case config[:url] do
-      webhook_url when is_binary(webhook_url) ->
-        case ExGram.set_webhook(
-               "https://#{webhook_url}/telegram/#{token_hash(token)}",
-               [{:token, token} | params]
-             ) do
-          {:ok, _} -> nil
-          {:error, error} -> Logger.error("Could not set the webhook: #{inspect(error)}")
-        end
+    case valid_url(config[:url]) do
+      {:ok, webhook_url} ->
+        ExGram.set_webhook("#{webhook_url}/telegram/#{token_hash(token)}", [
+          {:token, token} | params
+        ])
 
-      nil ->
-        Logger.warning(
-          "The webhook_url is not set in the configuration. Please manually set the webhook using this method: https://core.telegram.org/bots/api#setwebhook"
+      {:error, error} ->
+        Logger.error(
+          "The webhook_url is wrong with reason: #{inspect(error)}. Please manually set the webhook using this method: https://core.telegram.org/bots/api#setwebhook or edit the config file!"
         )
     end
   end
 
+  defp valid_url(nil), do: {:error, :not_set}
+  defp valid_url(url), do: url |> URI.parse() |> do_valid_url()
+
+  defp do_valid_url(%URI{scheme: nil}), do: {:error, :scheme_not_set}
```

Comment:
I would love to keep the old behaviour too, in order to be backwards compatible

```suggestion

  defp do_valid_url(%URI{scheme: nil, host: nil, path: path}) when is_binary(path), do: {:ok, "https://#{path}"}  
  
  defp do_valid_url(%URI{scheme: nil}), do: {:error, :scheme_not_set}
```

Reply:
done